### PR TITLE
Updated filter and update methods

### PR DESCRIPTION
**Pull request by writinwaters**
### What problem does this PR solve?

_Briefly describe what this PR aims to solve. Include background context that will help reviewers understand the purpose of the PR._

Issue link:#1490

### Type of change

- [x] Documentation Update

### CODE REVIEW

File: docs/references/pysdk_api_reference.md [hunk 1]
```diff
@@ -1351,12 +1247,12 @@ Deletes rows from the table based on the specified condition.
...
 ### Parameters
 
-#### cond: `str` (not empty), *Optional*
+#### cond: `str` (non-empty), *Optional*
 
 A condition or filter that determines which rows to delete from the table. The parameter can be an expression, a function, or any other form of conditional logic that evaluates to `True` for the rows that should be deleted. If `cond` is not specified or set to `None`, the method will delete all rows in the table.
 
 :::tip NOTE
-- The `cond` parameter currently supports 'and' and 'or' logical expressions only. Additional expressions like 'between' and 'in' will be available in future updates.
+- The `cond` parameter currently supports 'and' and 'or' logical expressions only.
```
Comment (JinHai-CN) [inline docs/references/pysdk_api_reference.md:1255]: need to double check

File: docs/references/pysdk_api_reference.md [hunk 2]
```diff
@@ -1454,34 +1350,77 @@ table_obj.update("c1 > 2", [{"c2": 100, "c3": 1000}])
 ```python
 table_obj.output(columns)
 ```
-Specify the columns to display in the search output, or perform aggregation operations or arithmetic calculations. 
+Specifies the columns to display in the search results or performs aggregation or arithmetic operations on them. This method allows you to customize the output of your query by selecting specific columns, applying aggregation functions, or performing arithmetic operations on the specified columns.
```
Comment (JinHai-CN) [inline docs/references/pysdk_api_reference.md:1353]: Remove the first sentence

File: docs/references/pysdk_api_reference.md [hunk 3]
```diff
@@ -1454,34 +1350,77 @@ table_obj.update("c1 > 2", [{"c2": 100, "c3": 1000}])
...
 
-### Parameters
+:::tip NOTE
+This method specifies the projection columns for the current table but does not directly produce displayable data. To display the query results, use `output()` in conjunction with methods like `to_result()`, `to_df()`, `to_pl()`, `to_arrow()`, or others that materialize the data.
```
Comment (JinHai-CN) [inline docs/references/pysdk_api_reference.md:1356]: column, constant, function, and other expressions.

File: docs/references/pysdk_api_reference.md [hunk 4]
```diff
@@ -1454,34 +1350,77 @@ table_obj.update("c1 > 2", [{"c2": 100, "c3": 1000}])
...
-- max
-- sum
-- avg
+:::tip NOTE
+The list must contain at least one element. Empty lists are not allowed.
+:::
   
 ### Returns
 
 An `infinity.local_infinity.table.LocalTable` object in Python module mode or an `infinity.remote_thrift.table.RemoteTable` object in client-server mode.
```
Comment (JinHai-CN) [inline docs/references/pysdk_api_reference.md:1386]: How to materialize the return value?

File: docs/references/pysdk_api_reference.md [hunk 5]
```diff
@@ -1454,34 +1350,77 @@ table_obj.update("c1 > 2", [{"c2": 100, "c3": 1000}])
...
 
-Must not be empty. Supported aggragation functions:
+#### columns: `list[str]` (non-empty), *Required*
+
+A non-empty list of strings specifying the columns to include in the output. Each string in the list can represent:
+
+- A user-defined column name: Directly specify the names of the columns to include in the output, e.g., `"body"`.
+- All user-defined columns: Use a wildcard `"*"` to select all columns.
+- A special system column: system-generated columns include:
+  - `_row_id`
```
Comment (JinHai-CN) [inline docs/references/pysdk_api_reference.md:1368]: Need to explain:
_row_id, auto generated, row_id in system might not aligned with actual row number of user. Once the data of the row is updated, row_id might also be changed.

File: docs/references/pysdk_api_reference.md [hunk 6]
```diff
@@ -1454,34 +1350,77 @@ table_obj.update("c1 > 2", [{"c2": 100, "c3": 1000}])
...
+  - `_score`
```
Comment (JinHai-CN) [inline docs/references/pysdk_api_reference.md:1369]: Used by full-text search, BM25 score.

File: docs/references/pysdk_api_reference.md [hunk 7]
```diff
@@ -1454,34 +1350,77 @@ table_obj.update("c1 > 2", [{"c2": 100, "c3": 1000}])
...
+  - `_similarity`
```
Comment (JinHai-CN) [inline docs/references/pysdk_api_reference.md:1370]: similarity, used by IP and cosine metric of dense vector. What about sparse and tensor?

File: docs/references/pysdk_api_reference.md [hunk 8]
```diff
@@ -1454,34 +1350,77 @@ table_obj.update("c1 > 2", [{"c2": 100, "c3": 1000}])
...
+  - `_distance`
```
Comment (JinHai-CN) [inline docs/references/pysdk_api_reference.md:1371]: L2 will use distance.

File: docs/references/pysdk_api_reference.md [hunk 9]
```diff
@@ -1454,34 +1350,77 @@ table_obj.update("c1 > 2", [{"c2": 100, "c3": 1000}])
...
+- An aggregation function: Apply an aggregation operation on specified columns. Supported aggragation functions include:
+  - `count`
+  - `min`
+  - `max`
+  - `sum`
+  - `avg`
+- An arithmetic function: Apply an arithmetic operation on specified columns (e.g., `c1+5`)
```
Comment (JinHai-CN) [inline docs/references/pysdk_api_reference.md:1378]: Other operator need to provide.

File: docs/references/pysdk_api_reference.md [hunk 10]
```diff
@@ -1454,34 +1350,77 @@ table_obj.update("c1 > 2", [{"c2": 100, "c3": 1000}])
...
 
 An `infinity.local_infinity.table.LocalTable` object in Python module mode or an `infinity.remote_thrift.table.RemoteTable` object in client-server mode.
 
 ### Examples
 
+#### Select columns to display
+
 ```python
+# Select all columns
 table_obj.output(["*"])
```
Comment (JinHai-CN) [inline docs/references/pysdk_api_reference.md:1394]: to_result

File: docs/references/pysdk_api_reference.md [hunk 11]
```diff
@@ -1490,24 +1429,36 @@ table_obj.output(["c1+5"])
 table_obj.filter(cond)
 ```
 
-Creates a filtering condition expression.
+Creates a filtering condition expression for the current table. This method allows you to specify conditions to filter rows based on the values in the columns.
```
Comment (JinHai-CN) [inline docs/references/pysdk_api_reference.md:1432]: Usually

File: docs/references/pysdk_api_reference.md [hunk 12]
```diff
@@ -1490,24 +1429,36 @@ table_obj.output(["c1+5"])
...
+
+:::tip NOTE
```
Comment (JinHai-CN) [inline docs/references/pysdk_api_reference.md:1434]: Use to describe return value.

File: docs/references/pysdk_api_reference.md [hunk 13]
```diff
@@ -1490,24 +1429,36 @@ table_obj.output(["c1+5"])
...
+This method specifies a filtering condition for the rows in the current table but does not directly produce displayable data. To display the query results, use `filter()` in conjunction with methods like `to_result()`, `to_df()`, `to_pl()`, `to_arrow()`, or others that materialize the data.
+:::
 
 ### Parameters
 
-#### cond: `str`
-`cond` has only supported 'and' and 'or' conjunction expression for now. 
+#### cond: `str` (non-empty), *Required*
+
+A non-empty string representing the filter condition. It comprises one or multiple expressions combined by 'and' or 'or' logical operators, where each expression uses comparison operators to set criteria for keeping or removing rows.
```
Comment (JinHai-CN) [inline docs/references/pysdk_api_reference.md:1442]: Lots of expressions can be used here.

File: docs/references/pysdk_api_reference.md [hunk 14]
```diff
@@ -1490,24 +1429,36 @@ table_obj.output(["c1+5"])
...
+Currently, only 'and' and 'or' logical expressions are supported.
+:::
 
 ### Returns
 
-- Success: self `Table`
-- Failure: `Exception`
+An `infinity.local_infinity.table.LocalTable` object in Python module mode or an `infinity.remote_thrift.table.RemoteTable` object in client-server mode.
 
 ### Examples
```
Comment (JinHai-CN) [inline docs/references/pysdk_api_reference.md:1453]: more examples:
1. include a function.
2. math expression.

File: docs/references/pysdk_api_reference.md [hunk 15]
```diff
@@ -1516,13 +1467,13 @@ table_obj.filter("(-7 < c1 or 9 >= c1) and (c2 = 3)")
 table_obj.knn(vector_column_name, embedding_data, embedding_data_type, distance_type, topn, knn_params = None)
 ```
 
-Conducts a KNN vector search, finding the top n closet rows to the given vector.
+Conducts a k-nearest neighbor (KNN) vector search, finding the top n closest rows to the given vector.
```
Comment (JinHai-CN) [inline docs/references/pysdk_api_reference.md:1470]: ANN need to be included.

Comment (JinHai-CN) [reply docs/references/pysdk_api_reference.md:1470]: For dense vector / embedding

File: docs/references/pysdk_api_reference.md (hunk 1 above)
Comment (writinwaters) [reply docs/references/pysdk_api_reference.md:1255]: BETWEEN AND and IN are not implemented yet.

File: docs/references/pysdk_api_reference.md [hunk 16]
```diff
@@ -681,13 +647,12 @@ An `IndexInfo` structure contains three fields,`column_name`, `index_type`, and
       - `"raw"`: Store the block-max index without compression.
 
 :::tip NOTE
-- Import the `infinity.index` package to set `IndexInfo`, `IndexType`, and `InitParameter`.
-- Import the `infinity` package to set `index.IndexInfo()`.
+Import the `infinity.index` package to set `IndexInfo`, `IndexType`, and `InitParameter`.
 
 ```python
+from infinity.index import IndexInfo
```
Comment (JinHai-CN) [inline docs/references/pysdk_api_reference.md:653]: from infinity.index import IndexInfo, IndexType, InitParameter

**Review by JinHai-CN — APPROVED** [review]

**Pull request merged**